src/main/java/com/geekbeast/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java

```diff
@@ -47,231 +30,82 @@
             return new SerializerConfig().setTypeClass( e.getKey() ).setImplementation( e.getValue() );
         } );
     }
 
+    @Override
     protected Map<String, MapConfig> getMapConfigs() {
         return Maps.transformEntries( mapRegistry, ( k, v ) -> {
             return v.getMapConfig();
         } );
     }
 
+    @Override
     protected Map<String, QueueConfig> getQueueConfigs() {
         return Maps.transformEntries( queueRegistry, ( k, v ) -> {
             return v.getQueueConfig();
         } );
     }
 
-    @Inject
+    /*
+     * The following three methods use @Autowired instead of @Inject, since that functionality is not provided by the
+     * spec :-/
+     */
+
+    @Autowired(
+        required = false )
     public void registerMapStores( Set<SelfRegisteringMapStore<?, ?>> mapStores ) {
+        if ( mapStores.isEmpty() ) {
+            logger.warn( "No map stores were configured." );
+        }
         for ( SelfRegisteringMapStore<?, ?> s : mapStores ) {
-            if ( s != null ) {
-                mapRegistry.put( s.getMapConfig().getName(), s );
-            } else {
-                logger.warn( "Encountered null entry when registering map stores." );
-            }
+
+            register( s.getMapConfig().getName(), s );
         }
     }
 
-    @Inject
+    @Autowired(
+        required = false )
     public void registerQueueStores( Set<SelfRegisteringQueueStore<?>> queueStores ) {
+        if ( queueStores.isEmpty() ) {
+            logger.warn( "No queue stores were configured." );
+        }
         for ( SelfRegisteringQueueStore<?> s : queueStores ) {
             queueRegistry.put( s.getQueueConfig().getName(), s );
         }
     }
 
-    @Inject
+    @Autowired(
+        required = false )
     public void register( Set<SelfRegisteringStreamSerializer<?>> serializers ) {
+        if ( serializers.isEmpty() ) {
+            logger.warn( "No serializers were configured." );
+        }
         for ( SelfRegisteringStreamSerializer<?> s : serializers ) {
             serializerRegistry.put( s.getClazz(), s );
         }
     }
 
     public static void register( String queueName, SelfRegisteringQueueStore<?> queueStore ) {
+        Preconditions.checkNotNull( queueStore, "Cannot register null queue-store." );
         queueRegistry.put( queueName, queueStore );
     }
 
     public static void register( String mapName, SelfRegisteringMapStore<?, ?> mapStore ) {
+        Preconditions.checkNotNull( mapStore, "Cannot register null map-store." );
         mapRegistry.put( mapName, mapStore );
     }
 
     public static void register( Class<?> hzSerializableClass, Serializer serializer ) {
+        Preconditions.checkNotNull( serializer, "Cannot register null serializer." );
         serializerRegistry.put( hzSerializableClass, serializer );
     }
 
-    // TODO: Make map and serializer beans as startup will fail if there is not at least one bean of that type.
```

Comment:
So the JSR-330 folks had a big argument about whether to allow optional injects and they decided that the "correct" way to to use providers.

Both Spring & Guava implemented their own optional injection, so I switched to the spring version of `@Inject` which is `@Autowired` and set `required=false`